io/eolearn/io/local_io.py

```diff
@@ -20,17 +20,20 @@ class ExportToTiff(SaveToDisk):
     :type folder: str
     :param band_count: Number of bands to be added to tiff image
     :type band_count: int
+    :param date_count: Number of dates to be added to tiff image
+    :type date_count: int
     :param image_dtype: Type of data to be saved into tiff image
     :type image_dtype: numpy.dtype
     :param no_data_value: Value of pixels of tiff image with no data in EOPatch
     :type no_data_value: int or float
     """
 
-    def __init__(self, feature, folder='.', *, band_count=1, image_dtype=np.uint8, no_data_value=0):
+    def __init__(self, feature, folder='.', *, band_count=1, date_count=1, image_dtype=np.uint8, no_data_value=0):
```

Comment:
I have 2 suggestions about `date_count` parameter:
- Instead of word `date` in the name it might be better to use word `time` as timestamps are not only dates but also datetimes.
- Instead of just a count this parameter could maybe also accept a time interval? Then it could select all times which fall into given time interval. But then the name would have to change to something more general...

Reply:
I agree on this point, but then you should allow the setting of intervals for `band_count` as well. Currently, you are basically only able to retrieve the `n` first bands, whereas you should be able to retrieve the `n` last bands too, or any other single band or intervals of bands. So maybe an argument in form of an index like `[n]` , `[-n]` or `[n:n+3]` would make the most sense.

Reply:
Yes, the `band_count` parameter could be improved as well. Maybe the following options could be accepted:
- A single integer `n` - it takes first `n` bands
- A tuple of 2 integers `(a, b)` - it takes bands `a, a+1, ..., b-1`
- A list of indices, e.g. `[0, 3, -4, -1]` - it takes bands with exactly these indices.
And something similar could be done with the time parameter. 

I hope I am not complicating too much. So far we were using this task only for exporting single channel timeless data and masks. But you never know when something more complicated will be required.

Reply:
I would say that for `time_count`, giving a `time_interval` makes most sense because users typically know roughly which date they need to retrieve, but likely not the exact time, unless they print the `patch.timestamp` attribute. 

For `band_count`, I would say all three options listed remain useful, as all combinations may be wanted. However, for the tuple option `(a, b)` I would extend the list to `a, a+1, ..., b-1, b` as one would expect both bounds to be inclusive.